### ISSUE TITLE
Estimated quantile loss too optimistic

### ISSUE THREAD

**Issue by arielf** [VowpalWabbit/vowpal_wabbit#370]
When using `--loss_function quantile` on a label with a mean and median of zero, I've been seeing final 'average loss' numbers that look too good to be true.  So I tried to reduce this to a trivial example.

Consider the following  (deliberately self-contradictory) data set where the label mean is zero:

```
-10 | a
10 | a
```

Since this is self contradictory it should not be converging, and the loss for the midpoint prediction cannot be smaller than `abs(0 - 10) = 10`.

But I get:

```
vw --loss_function quantile -c --passes 2000 --holdout_off quantile-loss.dat
... [progressive loss convergence seen here] ...

average loss = 3.75393
```

When looking at `loss_functions.cc` I see no `abs()` applied anywhere to the (predicted - actual) number

Is this a bug in the calculation of the overall amortized average  loss?

Shouldn't there be some `abs()` on the (predicted - actual) loss, when estimating the overall average loss, in quantile loss to prevent negative losses from cancelling out positive ones?  Note that in squared loss, we square the individual errors, so we never get negative vs positive cancellation.

On an additional level: the error is not simply canceling out.  If it were, I would expect it to converge all the way to zero error but it doesn't, so there's something more complex going on here.
